### PR TITLE
Log record info with uw-reopening ETL warnings

### PR DESCRIPTION
**Pull request by jccraft-uw**
Currently, when we see a warning logged from the UW Reopening ETL, we have
to go to the syslog and grep through to find the warning entry and then
find the associated record number. This takes a lot of time. This commit
logs the record ID and in some places the record instance ID for record
related warnings.

In the future when we have a structured logging solution, we will have a
different way to get the record context with the event.

### CODE REVIEW

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py [hunk 1]
```diff
@@ -137,12 +137,14 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
...
-        LOG.warning("Skipping record with insufficient information to construct patient")
+        LOG.warning("Skipping record with insufficient information to construct patient for record: "
+            f"{enrollment.get('record_id')}")
         return None
 
     birthdate = parse_date_from_string(enrollment.get('core_birthdate'))
     if not birthdate:
-        LOG.warning("Record has an invalid or missing `core_birthdate` value")
+        LOG.warning("Record has an invalid or missing `core_birthdate` value for record: "
+            f"{enrollment.get('record_id')}")
```
Comment (tsibley) [inline lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py:147]: Very minor, but the convention in other log messages in our codebase puts the id immediately adjacent to the noun, e.g.:

```python
LOG.warning(f"Record {enrollment.get('record_id')} has an invalid or missing `core_birthdate` value")
```

so that the message is more concise and reads as a natural sentence.

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py [hunk 2]
```diff
@@ -171,7 +173,8 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
             elif is_complete('test_order_survey', redcap_record_instance):
                 collection_method = CollectionMethod.SWAB_AND_SEND
         else:
-            LOG.warning(f"The record instance has an unexpected event name: {redcap_record_instance.event_name}")
+            LOG.warning(f"The record instance has an unexpected event name: '{redcap_record_instance.event_name}' "
```
Comment (tsibley) [inline lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py:176]: FWIW, using `{redcap_record_instance.event_name!r}` will produce a quoted string that's always unambiguous regardless of contents, unlike surrounding the interpolation with literal quotes.

**Review by tsibley — APPROVED** [review]

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py (hunk 1 above)
Comment (jccraft-uw) [reply lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py]: That was a bit wordy! Fixed.

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py (hunk 2 above)
Comment (jccraft-uw) [reply lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py]: Oh cool. I saw the "!r" in something you wrote this week and didn't dig into what it does ... 'r' being the internal representation ... repr.

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py [hunk 3]
```diff
@@ -137,12 +137,13 @@ def redcap_det_uw_reopening(*, db: DatabaseSession, cache: TTLCache, det: dict,
             system_identifier = INTERNAL_SYSTEM)
 
     if not patient_entry:
-        LOG.warning("Skipping record with insufficient information to construct patient")
+        LOG.warning("Skipping record with insufficient information to construct patient for record: "
+            f"{enrollment.get('record_id')}")
```
Comment (kschwabefry) [inline lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py:141]: The same convention could be applied here, like: 
"Skipping record {enrollment.get('record_id')} with insufficient information to construct patient"

**Review by kschwabefry — APPROVED** [review]
Having these details will make debugging easier. Thanks for adding!

File: lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py (hunk 3 above)
Comment (jccraft-uw) [reply lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py]: Changed!

**Comment by jccraft-uw** [comment]
Deployed.
